indico/modules/events/management/controllers/cloning.py

```diff
@@ -196,3 +198,62 @@ def _process(self):
                         for c in EventCloner.get_cloners(self.event)}
         return jsonify_template('events/management/clone_event.html', event=self.event, step=step, form=form,
                                 cloner_dependencies=dependencies, **tpl_args)
+
+
+class RHImportFromEvent(RHManageEventBase):
+    """Import data from another event."""
+
+    def _form_for_step(self, step, set_defaults=True):
+        if step == 1:
+            return ImportSourceEventForm()
+        elif step == 2:
+            source_event = get_event_from_url(request.form.get('source_event_url'))
+            return ImportContentsForm(source_event, self.event, set_defaults=set_defaults)
+        else:
+            return None
+
+    def _process(self):
+        step = int(request.form.get('step', 1))
+        form = self._form_for_step(step, set_defaults=True)
+        prev_form = self._form_for_step(step - 1)
+
+        if prev_form and not prev_form.validate():
+            form = prev_form
+            step = step - 1
+
+        elif step > 2:
+            # last step - perform actual cloning
+            source_event = get_event_from_url(request.form.get('source_event_url'))
+            form = ImportContentsForm(source_event, self.event)
+
+            if form.validate_on_submit():
+                updated_event = clone_into_event(source_event, self.event, set(form.selected_items.data))
+                flash(_('Data properly imported!'), 'success')
+                return jsonify_data(redirect=url_for('event_management.settings', updated_event), flash=False)
+            else:
+                # back to step 2, since there's been an error
+                step = 2
+        dependencies = {c.name: {'requires': list(c.requires_deep), 'required_by': list(c.required_by_deep)}
+                        for c in EventCloner.get_cloners(self.event)}
+        return jsonify_template('events/management/import_event.html', event=self.event, step=step, form=form,
+                                cloner_dependencies=dependencies)
+
+
+class RHImportEventDetails(RHManageEventBase):
+    ALLOW_LOCKED = True
+
+    def _process(self):
+        from indico.modules.events.schemas import EventDetailsSchema
+        schema = EventDetailsSchema()
+        form = ImportSourceEventForm()
+        try:
+            event = get_event_from_url(form.source_event_url.data)
+            if event == self.event:
+                raise ValueError(_('Cannot import from the same event'))
+            if event.type != self.event.type:
```

Comment:
```suggestion
            if event.type_ != self.event.type_:
```

`type_` is the enum, `type` is its name (for legacy reasons). No need to use the latter in new code ;)